blockchain-source/src/cardano/event.rs

```diff
@@ -105,50 +105,39 @@ impl<Tip> GetNextFrom for CardanoNetworkEvent<BlockEvent, Tip> {
 
 impl BlockEvent {
     pub(crate) fn from_serialized_block(raw_block: &[u8], era: &Era) -> anyhow::Result<Self> {
-        let block: anyhow::Result<crate::cardano::block::Block> =
-            cbored::decode_from_bytes(raw_block).context("failed to deserialize block");
-
-        if let Ok(block) = block {
-            let header = block.header();
-            let id = BlockId::new(header.hash().to_string());
-            let block_number = BlockNumber::new(header.block_number());
-
-            let parent_id = get_parent_id(&block.header());
-
-            Ok(BlockEvent {
-                raw_block: raw_block.to_vec(),
-                id,
-                parent_id,
-                block_number,
-                slot_number: SlotNumber::new(header.slot()),
-                is_boundary_block: false,
-                // this is not in the header, and computing it requires knowing the network
-                // details, which makes implementing `Serialize` and `Deserialize`more complicated,
-                // unless we serialize this field too.
-                // it can be computed later inside carp, since we don't need this in the bridge.
-                epoch: era
+        let block = cml_multi_era::MultiEraBlock::from_explicit_network_cbor_bytes(raw_block)
```

Comment:
it would be really nice if we could decode only the header here, because then the block is deserialized twice... But I guess this is not possible